### PR TITLE
Flipper white blacklist

### LINKED ISSUE #243: white or blacklist items
There should be an option to white or blacklist items.
Often users are only interested in flipping certain items or want to exclude others. 
Its up for debate if the white/blacklist should be tied to an account.
I suggest that hiding an item can be done via a custom right-click menu

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #243

### CODE REVIEW

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-412.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-412.flou21.de

File: src/api/HttpHelper.tsx [hunk 1]
```diff
@@ -44,48 +46,82 @@ function sendRequest(request: ApiRequest, cacheInvalidationGrouping?: number): P
...
     })
+}
 
-    function isResponseValid(response: Response) {
-        return response.ok && response.body;
-    }
+/**
+ * Sends API-Request to the backend
+ * @param request The request-Object
+ * @param cacheInvalidationGrouping A number which is appended to the url to be able to invalidate the cache
```
Comment (Ekwav) [inline src/api/HttpHelper.tsx:56]: This param doesn't exist

File: src/api/HttpHelper.tsx [hunk 2]
```diff
@@ -44,48 +46,82 @@ function sendRequest(request: ApiRequest, cacheInvalidationGrouping?: number): P
...
+ * @returns A emty promise (the resolve/reject Method of the request-Object is called)
+ */
+function sendApiRequest(request: ApiRequest): Promise<void> {
+    request.mId = getNextMessageId();
+    let requestString = request.data;
+    let headers = { 'ConId': getOrGenerateUUid(), 'mode': 'no-cors' };
```
Comment (Ekwav) [inline src/api/HttpHelper.tsx:62]: The ConId is useless if its generated every time

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-412.flou21.de

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env-sky-412.flou21.de

File: src/components/Flipper/Flipper.tsx [hunk 3]
```diff
@@ -261,8 +261,43 @@ function Flipper() {
...
+        }
+
+        if (filter?.restrictions !== undefined) {
+
+            let isValid = true;
+
+            filter.restrictions.forEach(restriction => {
+
+                if (restriction.type === "blacklist" && restriction.item && restriction.item.tag === flipAuction.item.tag) {
+                    isValid = false;
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:278]: you should break out of the loop on the first match.
Or even better just make it a normal for loop and return false (more efficient)

File: src/components/Flipper/Flipper.tsx [hunk 4]
```diff
@@ -284,7 +319,7 @@ function Flipper() {
     );
 
     let customizeFlipDialog = (
-        <Modal size={"xl"} show={showCustomizeFlip} onHide={() => { setShowCustomizeFlip(false) }}>
+        <Modal key={"test"} size={"xl"} show={showCustomizeFlip} onHide={() => { setShowCustomizeFlip(false) }}>
```
Comment (Ekwav) [inline src/components/Flipper/Flipper.tsx:322]: is this key intended to be here?

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Review by Ekwav — COMMENTED** [review]
thanks I hate it

**Comment by Flou21** [comment]
[CI] created a pr environment: https://pr-env.flou.dev/412

**Pull request merged**